### ISSUE TITLE
Getter and setter attributes are mutable

### ISSUE THREAD

**Issue by cpetrov** [eclipsesource/tabris-js#542]
This can sometimes lead to unexpected behavior. 

In this example modifying the layoutData returned by the colorfulComposite's `get` method leads to a modification in the widget configuration being applied.

```javascript
var page = tabris.create("Page", {topLevel: true}).open();

var configuration = {
  "#colorfulComposite": {
    background: "red",
    layoutData: {centerX: 0, centerY: 0, width: 200, height: 200}
  },
  "#enlargeButton": {
    layoutData: {centerX: 0, top: ["#colorfulComposite", 20]}
  }
}

var colorfulComposite = tabris.create("Composite", {id: "colorfulComposite"})
  .appendTo(page);
tabris.create("Button", {
  id: "enlargeButton",
  text: "Enlarge Composite"
}).on("select", function () {
  var colorfulCompositeLayoutData = colorfulComposite.get("layoutData");
  colorfulCompositeLayoutData.width *= 2;
  colorfulCompositeLayoutData.height *= 2;
  colorfulComposite.set("layoutData", colorfulCompositeLayoutData);
  console.log(JSON.stringify(configuration["#colorfulComposite"])); // {..., "layoutData": {..., "width": 400, "height": 400}}
}).appendTo(page);

page.apply(configuration);
```

**Comment by tbuschto** [comment]
We did a partial fix for 1.1. in commit 44290bf32235b0db9af83787e6451b00bfdfc945. This only does a shallow copy for layoutData. 

Possibly a better fix for these properties could be to treat them as shorthands, i.e. actually store layoutData as layoutData-left, layoutData-right, etc. With our nativeBridge caching that would not hurt performance, but It would complicate checks.

**Comment by tbuschto** [comment]
I have been looking at the enocode/decode functions for layoutData, transform, font, image and array, and by now they are all creating (sufficently deep) copies while setting/getting values.
